src/Admin/BaseFieldDescription.php

```diff
@@ -521,14 +521,14 @@ private function getFieldGetterKey($object, ?string $fieldName): ?string
         return implode('-', $components);
     }
 
-    private function hasCachedFieldGetter($object, string $fieldName): bool
+    private function hasCachedFieldGetter(object $object, string $fieldName): bool
```

Comment:
How is it infer ?

The three methods are calling `getFieldGetterKey`. Which does a
```
if (!\is_object($object)) {
```
check

Reply:
I suppose it looks at all the calls in the codebase. I think this check is in case `$object` is `null`, which it can't be at the only call site for `hasCachedFieldGetter` in the codebase:

https://github.com/sonata-project/SonataAdminBundle/blob/fa11a60ab10739529dc60780f71a80205ede12b6/src/Admin/BaseFieldDescription.php#L311-L328

Reply:
Indeed, so we should update `getFieldGetterKey` typehint too, and remove the object check in the method.

Reply:
If I run Psalm `--alter` again, it does it by itself, because it can infer more things now 👍

Reply:
Psalm didn't remove the 
```
if (!\is_object($object)) {
    return null;
}
```

Part ;)

Reply:
Oh right, it only reports it.